File: src/RehabEasy.Domain/Models/ApiPayloadImportResult.cs
namespace RehabEasy.Domain.Models;

public sealed class ApiPayloadImportResult
{
    public string PayloadId { get; init; } = string.Empty;
    public string SourceName { get; init; } = string.Empty;
    public DateTimeOffset ImportedAt { get; init; }
    public string? PdfUrl { get; init; }
    public string PdfLocalPath { get; init; } = string.Empty;
    public IReadOnlyList<RehabEasyRecord> Records { get; init; } = Array.Empty<RehabEasyRecord>();
}

File: java/src/main/java/com/rehabeasy/model/ApiPayloadImportResult.java
package com.rehabeasy.model;

import java.time.Instant;
import java.util.List;

public record ApiPayloadImportResult(
        String payloadId,
        String sourceName,
        Instant importedAt,
        String pdfUrl,
        String pdfLocalPath,
        List<RehabEasyRecord> records
) {
    public ApiPayloadImportResult {
        payloadId = payloadId == null ? "" : payloadId;
        sourceName = sourceName == null ? "" : sourceName;
        importedAt = importedAt == null ? Instant.EPOCH : importedAt;
        records = records == null ? List.of() : List.copyOf(records);
    }
}
